ssl/t1_lib.c

```diff
@@ -1706,9 +1957,12 @@ int tls1_set_server_sigalgs(SSL_CONNECTION *s)
     OPENSSL_free(s->shared_sigalgs);
     s->shared_sigalgs = NULL;
     s->shared_sigalgslen = 0;
-    /* Clear certificate validity flags */
-    for (i = 0; i < SSL_PKEY_NUM; i++)
-        s->s3.tmp.valid_flags[i] = 0;
+
+    /* Allocate and clear certificate validity flags */
+    OPENSSL_free(s->s3.tmp.valid_flags);
+    s->s3.tmp.valid_flags = OPENSSL_zalloc(s->ssl_pkey_num * sizeof(uint32_t));
```

Comment:
Could this be a memset if it is already allocated?

Reply:
Yes, so changed in 0ac814f8247d1a5f157eec2b62bce0000c3a2d1e.